src/pg_ext/big_decimal.cr

```diff
@@ -0,0 +1,23 @@
+require "big"
+
+module PG
+  struct Numeric
+    # Returns a BigDecimal representation of the numeric. This retains all precision.
+    def to_big_d
+      return BigDecimal.new("0") if nan? || ndigits == 0
+
+      # Since BigDecimal allows initialaztion from String, why should one reinvent the wheel?
+      BigDecimal.new(to_s)
+    end
+  end
+
+  class ResultSet
+    def read(t : BigDecimal.class)
+      read(PG::Numeric).to_big_d
+    end
+
+    def read(t : BigDecimal?.class)
+      read(PG::Numeric?).try &.to_big_d
```

Comment:
Is there a way to avoid `try` here?
```crystal
(r = read(PG::Numeric?)) ? r.to_big_d : nil
```

Reply:
Why would you want to avoid `try`?

Reply:
After looking into Crystal's implementation, I retract my comment (at least for this code).

In Rails, `try` is one of the more abused methods - often chained together, usually indicating a poor design. It just became a habit for me to avoid it altogether.